### PR TITLE
Add resources for Mi 8 series, Redmi 6(A), Redmi Note 6 Pro and Redmi S2

### PR DESCRIPTION
**Pull request by suwakowww**
Rewrote #39, the original branch is too mess to commit, sorry.

For:
* dipper ([Referrer](https://github.com/TadiT7/xiaomi_dipper_dump/blob/dipper-user-8.1.0-OPM1.171019.011-V9.5.13.0.OEAMIFA-release-keys/system/framework/framework-res.apk), tested and working)
* ursa ([Referrer](https://github.com/TadiT7/xiaomi_ursa_dump/blob/ursa-user-8.1.0-OPM1.171019.026-V9.6.5.0.OEHCNFD-release-keys/system/framework/framework-res.apk), blind-patched)
* equuleus ([Referrer](https://github.com/TadiT7/xiaomi_equuleus_dump/blob/equuleus-user-8.1.0-OPM1.171019.026-V9.6.6.0.OECCNFD-release-keys/system/framework/framework-res.apk)<strike>, notch same as ursa</strike>, blind-patched, maybe fix for phhusson/treble_experimentations#336)
* platina ([Referrer](https://github.com/TadiT7/xiaomi_platina_dump/blob/platina-user-8.1.0-OPM1.171019.019-V9.6.6.0.ODTCNFD-release-keys/system/framework/framework-res.apk), tested by `用户名需大于5位小于 @ Coolapk`)
* cereus ([Referrer](https://github.com/bin8001/dump_xiaomi_cereus/blob/master/system/framework/framework-res.apk), blind-patched)
* cactus ([Referrer](https://github.com/TadiT7/xiaomi_cactus_dump/blob/cactus-user-8.1.0-O11019-V9.6.2.0.OCBMIFD-release-keys/system/framework/framework-res.apk), blind-patched)
* ysl ([Referrer](https://github.com/TadiT7/xiaomi_ysl_dump/blob/CORRUPTED---ysl-user-8.1.0-OPM1.171019.011-8.6.21-release-keys/system/framework/framework-res.apk), blind-patched)
* tulip ([Referrer](https://github.com/TadiT7/xiaomi_tulip_dump/blob/tulip-user-8.1.0-OPM1.171019.011-V9.6.10.0.OEKMIFD-release-keys/system/framework/framework-res.apk), blind-patched, maybe fix for phhusson/treble_experimentations#300 and phhusson/treble_experimentations#301)

### CODE REVIEW

**Comment by TadiT7** [comment]
Why not referring to Pie?

**Comment by suwakowww** [comment]
>           Why not referring to Pie?

On Android 9.0, there's some keys has been changed on power_profile.xml, I tried to made for this on my wayne but got bootloop: phhusson/vendor_hardware_overlay@dd9479c

Referrer: 
cpu.speed --> cpu.core_speed: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L349
cpu.active --> cpu.core_power: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L350
dsp.audio --> audio: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L181
dsp.video --> video: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L187

And, it used to be:
cpu.speed: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L185
cpu.active: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L59
dsp.audio: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L156
dsp.video: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L162

**Comment by TadiT7** [comment]
> > ```
> >       Why not referring to Pie?
> > ```
> 
> On Android 9.0, there's some keys has been changed on power_profile.xml, I tried to made for this on my wayne but got bootloop: [dd9479c](https://github.com/phhusson/vendor_hardware_overlay/commit/dd9479c)
> 
> Referrer:
> cpu.speed --> cpu.core_speed: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L349
> cpu.active --> cpu.core_power: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L350
> dsp.audio --> audio: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L181
> dsp.video --> video: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/com/android/internal/os/PowerProfile.java#L187
> 
> And, it used to be:
> cpu.speed: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L185
> cpu.active: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L59
> dsp.audio: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L156
> dsp.video: https://github.com/aosp-mirror/platform_frameworks_base/blob/oreo-mr1-release/core/java/com/android/internal/os/PowerProfile.java#L162

So I suggest to keep Oreo power profiles, but use Pie framework values. Pushed Global Dev Pie for equuleus today and about to push platina Pie.

**Comment by TadiT7** [comment]
Platina Pie pushed, this pr will be perfect if you use differ.sh on Pie

**Comment by phhusson** [comment]
(I don't know why the bot didn't test this, but I checked it and it passes tests)

File: Xiaomi/Mi8/res/values/config.xml [hunk 1]
```diff
@@ -103,13 +103,13 @@
     <fraction name="config_autoBrightnessAdjustmentMaxGamma">200.0%</fraction>
     <fraction name="config_maximumScreenDimRatio">29.999996%</fraction>
     <integer name="config_autoBrightnessBrighteningLightDebounce">2000</integer>
-    <integer name="config_autoBrightnessDarkeningLightDebounce">6000</integer>
+    <integer name="config_autoBrightnessDarkeningLightDebounce">2000</integer>
     <integer name="config_screenBrightnessDark">1</integer>
-    <integer name="config_screenBrightnessDim">1</integer>
+    <integer name="config_screenBrightnessDim">4</integer>
```
Comment (phhusson) [inline Xiaomi/Mi8/res/values/config.xml]: Dim is no higher than config_screenBrightnessSettingMinimum
That sounds wrong.
Is this stock ROM value?

File: Xiaomi/Mi8/res/values/config.xml [hunk 2]
```diff
@@ -103,13 +103,13 @@
     <fraction name="config_autoBrightnessAdjustmentMaxGamma">200.0%</fraction>
     <fraction name="config_maximumScreenDimRatio">29.999996%</fraction>
     <integer name="config_autoBrightnessBrighteningLightDebounce">2000</integer>
-    <integer name="config_autoBrightnessDarkeningLightDebounce">6000</integer>
+    <integer name="config_autoBrightnessDarkeningLightDebounce">2000</integer>
```
Comment (phhusson) [inline Xiaomi/Mi8/res/values/config.xml]: This new value comes from stock ROM? Where did the old value come from?

File: Xiaomi/Mi8/res/values/config.xml (hunk 1 above)
Comment (suwakowww) [reply Xiaomi/Mi8/res/values/config.xml]: Both of them are come from stock ROM but different Android version.
`1` froms Oreo based ROM, and `4` froms Pie based ROM
But I think I should changing it to `2`

File: Xiaomi/Mi8/res/values/config.xml (hunk 2 above)
Comment (suwakowww) [reply Xiaomi/Mi8/res/values/config.xml]: Same as the reply of last review.
`6000` froms Oreo based ROM, and `2000` froms Pie based ROM.

**Pull request merged**